test/integration/dialects/abstract/connection-manager.test.ts

```diff
@@ -0,0 +1,186 @@
+        read: [{ ...poolEntry }, { ...poolEntry }],
+      },
+    };
+
+    const sequelize = createSequelizeInstance(options);
+    expect(sequelize.connectionManager.pool).to.be.instanceOf(ReplicationPool);
+    expect(sequelize.connectionManager.pool.read).to.be.instanceOf(Pool);
+    expect(sequelize.connectionManager.pool.write).to.be.instanceOf(Pool);
+  });
+
+  it('should round robin calls to the read pool', async () => {
+    if (getTestDialect() === 'sqlite') {
+      return;
+    }
+
+    const slave1 = { ...poolEntry };
+    const slave2 = { ...poolEntry };
+    slave1.host = 'slave1';
+    slave2.host = 'slave2';
+
+    const options = {
+      replication: {
+        write: { ...poolEntry },
+        read: [slave1, slave2],
+      },
+    };
+
+    const sequelize = createSequelizeInstance(options);
+    const connectionManager = sequelize.connectionManager;
+
+    const res: Connection = {};
+
+    const connectStub = sandbox.stub(connectionManager, '_connect').resolves(res);
+    sandbox.stub(connectionManager, '_disconnect').resolves();
+    sandbox.stub(sequelize, 'databaseVersion').resolves(sequelize.dialect.defaultVersion);
+
+    const queryOptions: GetConnectionOptions = {
+      type: 'read',
+      useMaster: false,
+    };
+
+    const _getConnection = connectionManager.getConnection.bind(
+      connectionManager,
+      queryOptions,
+    );
+
+    await _getConnection();
+    await _getConnection();
+    await _getConnection();
+    chai.expect(connectStub.callCount).to.equal(4);
+
+    // First call is the get connection for DB versions - ignore
+    const calls = connectStub.getCalls();
+    chai.expect(calls[1].args[0].host).to.eql('slave1');
+    chai.expect(calls[2].args[0].host).to.eql('slave2');
+    chai.expect(calls[3].args[0].host).to.eql('slave1');
+  });
+
+  // sqlite does not have different database versions since it's bundled
+  if (dialect !== 'sqlite') {
+    it('should trigger deprecation for non supported engine version', async () => {
+      const stub = sandbox.stub(process, 'emitWarning');
+      const sequelize = createSequelizeInstance();
+      const connectionManager = sequelize.connectionManager;
+
+      sandbox.stub(sequelize, 'databaseVersion').resolves('0.0.1');
+
+      const res: Connection = {};
+
+      sandbox.stub(connectionManager, '_connect').resolves(res);
+      sandbox.stub(connectionManager, '_disconnect').resolves();
+
+      const queryOptions: GetConnectionOptions = {
+        type: 'read',
+        useMaster: true,
+      };
+
+      await connectionManager.getConnection(queryOptions);
+      chai.expect(stub).to.have.been.calledOnce;
+      chai
+        .expect(stub.getCalls()[0].args[0])
+        .to.contain(
+          'This database engine version is not supported, please update your database server.',
+        );
+      stub.restore();
+    });
+  }
+
+  if (dialect !== 'sqlite') {
+    // sqlite overrides getConnection() and never uses _connect
```

Comment:
Is this a change? Because the test did run on sqlite before right?

Reply:
The old test suite [created its own ConnectionManager](https://github.com/sequelize/sequelize/blob/main/test/integration/dialects/abstract/connection-manager.test.js#L164-L167) instead of using the current dialect's.

So this was not really tested for any dialect except 'abstract', so no dialect